gdal/ogr/ogrsf_frmts/ods/ogrodsdatasource.cpp

```diff
@@ -657,29 +665,33 @@ void OGRODSDataSource::startElementTable(const char *pszNameIn,
     {
         nRowsRepeated = atoi(
             GetAttributeValue(ppszAttr, "table:number-rows-repeated", "1"));
-        if (nRowsRepeated < 0 || nRowsRepeated > 10000)
+        
+        if (nRowsRepeated < 0)
         {
-            CPLError(CE_Failure, CPLE_NotSupported,
-                     "Invalid value for number-rows-repeated = %d",
-                     nRowsRepeated);
-            bEndTableParsing = true;
-            return;
+          nRowsRepeated = 1;
         }
-        const int nFields = std::max(
-            static_cast<int>(apoFirstLineValues.size()),
-            poCurLayer != nullptr ?
-                poCurLayer->GetLayerDefn()->GetFieldCount() : 0);
-        if( nFields > 0 && nRowsRepeated > 100000 / nFields )
+        
+        nRowsRepeated = std::min(nRowsRepeated, 10000);
```

Comment:
This will silently clamp number-rows-repeated. I prefered the previous explicit error strategy